### DISCUSSION TITLE
Custom Report

### DISCUSSION THREAD

**Discussion by fbarbeau** [kimai/kimai#4332] [Questions & Answers]
Hi,
is it possible to create custom report?

We would like to have the infos from the individual user report but for all the users. When I do the all users report I only have the total time of the day. I would like to see the break down of that day.

Thanks,

**Comment by lucas-0liveira** [comment]
Yes, it's definitely possible to create a custom report for that. We can compile the information from the individual user reports and provide you with a detailed breakdown for all users. This way, you'll have a comprehensive view of the activities throughout the day. Just let us know what specific details you'd like to include in the report, and we'll make it happen!

**Reply by kevinpapst** [reply]
Stop posting ChatGPT generated non-sense or I will block you.

This is related to the hidden comment, not the original post

**Comment by fbarbeau** [comment]
@kevinpapst Hi, I'm not sure I understand the ChatGPT part of your comment? Is it possible to have custom report or not?

Thanks,

**Comment by derStephan** [comment]
Is there some documentation on how to add own reports?

**Reply by kevinpapst** [reply]
https://github.com/kimai/DemoBundle/releases/tag/2.4.0

**Reply by kevinpapst** [reply]
I added a demo report to the DemoBundle, so you can see how it is done.
Here is the commit:
https://github.com/kimai/DemoBundle/commit/a3e7d1e80fea0cb2cf17bc2e5ddf3df24497ae84

**Reply by derStephan** [reply]
Thanks for that. I tried to understand how these things play together as I am not familiar with symfony. Can you give some example, where I would do a query on the database? Or is there any abstraction layer that I am missing?

**Reply by kevinpapst** [reply]
Kimai uses Symfony and adds its own API on top. 
So I would suggest to read their documentation and then come back with detail questions. 
If the demo bundle doesn't include an example for your use-case, I can probably add one.

**Reply by derStephan** [reply]
That would be very nice. I just want to have the data of the contract page inluding overtime in a report. So I thought, I can just assemble some SQL as [given here ](https://www.kimai.org/documentation/sql-examples.html) but I don't know in which file I have to put it. 

So for me it would be perfect if the demo bundle included some example SQL query including communication with the database and showing the results in a table.

**Reply by derStephan** [reply]
Do you know when you can spare some time for an example?
